### PR TITLE
FEATURE: Render help message from yaml file

### LINKED ISSUE #1694: Neos does not render Help message from Yaml file.
The Help message is not rendered by following the guide at http://neos.readthedocs.io/en/3.3/References/PropertyEditorReference.html#property-type-string-textfieldeditor-single-line-text-editor-default.

(Taken from https://github.com/neos/neos-development-collection/issues/1830 @Nguonchhay)

### PR DESCRIPTION
**Pull request by mstruebing**
This currently shows a help icon in the NodeCreateionDialog and inspector property if a help definition exists in the yaml file and sets the title of this element to the help message.
Thus means if you hover this item you will see the help message.

closes  #1694

New:
screenshot_2018-06-27 viii the queen s croquet-ground - the book - neos demo site
screenshot_2018-06-27 viii the queen s croquet-ground - the book - neos demo site 1

Old:
screenshot_2018-06-27 viii the queen s croquet-ground - the book - neos demo site 2
screenshot_2018-06-27 viii the queen s croquet-ground - the book - neos demo site 3

### CODE REVIEW

File: packages/neos-ui-editors/src/EditorEnvelope/index.js [hunk 1]
```diff
@@ -21,6 +23,7 @@ export default class EditorEnvelope extends PureComponent {
         editorRegistry: PropTypes.object.isRequired,
         validationErrors: PropTypes.array,
         onEnterKey: PropTypes.func,
+        helpMessage: PropTypes.string,
```
Comment (johannessteu) [inline packages/neos-ui-editors/src/EditorEnvelope/index.js]: Do we miss a default value here?

File: packages/neos-ui-editors/src/EditorEnvelope/index.js [hunk 2]
```diff
@@ -86,7 +102,9 @@ export default class EditorEnvelope extends PureComponent {
 
         return (
             <div>
-                {this.renderLabel()}
+                <span title={this.props.helpMessage}>
```
Comment (johannessteu) [inline packages/neos-ui-editors/src/EditorEnvelope/index.js]: Maybe a fallback to the label if no helpmessage is set?

**Comment by johannessteu** [comment]
The `thumbnail` is still missing, right?
http://neos.readthedocs.io/en/3.3/CreatingASite/NodeTypes/NodeTypeDefinition.html

File: packages/neos-ui-editors/src/EditorEnvelope/index.js (hunk 2 above)
Comment (mstruebing) [reply packages/neos-ui-editors/src/EditorEnvelope/index.js]: If there is no helpmessage there should be no helpmessage icon rendered, so there also shouldn't be a fallback IMO

File: packages/neos-ui-editors/src/EditorEnvelope/index.js (hunk 1 above)
Comment (mstruebing) [reply packages/neos-ui-editors/src/EditorEnvelope/index.js]: I would say that `helpMessage` is `undefined` and if we have an `undefined` `helpMessage` we do not render the help-icon and title

**Comment by mstruebing** [comment]
Jup `thumbnail` is currently missing.

**Comment by mstruebing** [comment]
Now the already present `Tooltip`-component is used.

Click-outside is missing, somehow I didn't get it to work and the thumbnail isn't scaled currently.

File: packages/neos-ui-editors/src/EditorEnvelope/index.js [hunk 3]
```diff
@@ -70,12 +75,40 @@ export default class EditorEnvelope extends PureComponent {
...
+        this.setState({
+            showTooltip: !this.state.showTooltip
+        });
+    };
+
+    renderTooltip() {
+        return (
+            <Tooltip className={style.envelope__tooltip}>
+                {this.props.helpMessage ? this.props.helpMessage : ''}
+                {this.props.helpThumbnail ? <img className={style.envelope__tooltip__image} src={this.props.helpThumbnail} /> : ''}
```
Comment (johannessteu) [inline packages/neos-ui-editors/src/EditorEnvelope/index.js]: Is the `envelope__tooltip__image` defined and acutally used?

File: packages/react-ui-components/src/Tooltip/tooltip.js [hunk 4]
```diff
@@ -37,7 +39,12 @@ Tooltip.propTypes = {
     /**
      * An optional css theme to be injected.
      */
-    theme: PropTypes.object
+    theme: PropTypes.object,
+
+    /**
+     * Whether this tooltip should indicate an error or not
+     */
+    error: PropTypes.bool
```
Comment (johannessteu) [inline packages/react-ui-components/src/Tooltip/tooltip.js]: not sure if a name like displayError or something fits better

**Comment by mstruebing** [comment]
Merging because @johannessteu needs to use this in a followup  PR

**Comment by jonnitto** [comment]
@mstruebing Is it possible that the help message doesn't work on a boolean editor? With this configuration, I get no help icon:

```yaml
Foo.Bar:Checkbox:
  properties:
    checkbox:
      type: boolean
      ui:
        label: Label
        help:
          message: Message
```

**Comment by mstruebing** [comment]
Can confirm this issue.Tested on the demo page with latest master right now.

**Pull request merged**